fafbot.py

```diff
@@ -310,72 +387,245 @@ def updateBalance(self, amount, uid):
+                #print "no uid"
+                return
+
+            playeruid = self.getUid(who)
+            if not playeruid:
+                #print "no player uid"
+                return
+
+            match = self.getMatches(playeruid)
+            if not match:
+                #print "no match"
+                return
+
+            match = self.betting.addMatch(match)
+            if not match:
+                #print "no match 2"
+                return
+
+            balance = self.currentBalance(uid)
+
+            if uid in match.players:
+                return
+
+            resultAmount = match.addBeter(amount, uid, playeruid)
+            if resultAmount < amount:
+                text = "%s has taken a bet of %i (amount reduced due to the in-game time) on \"%s\" (his balance is now %i)." % (source, resultAmount, match.name, balance-resultAmount)
+                self.connection.privmsg(source, text)
+                self.updateBalance(balance-resultAmount, uid)
+            else:
+                text = "%s has taken a bet of %i on \"%s\" (his balance is now %i)." % (source, amount, match.name, balance-amount)
+                self.connection.privmsg(source, text)
+                self.updateBalance(balance-amount, uid)
 
+    def mods_Msg(self,sender):
+        """
+        !mods - private messages sender a list of online mods
+        """
+        ignorelist = ["fafbot"]
+        channelObj = self.channels.items()[0][1]
+        mods = channelObj.opers() + channelObj.owners() + channelObj.halfops()
+        if mods:
+            self.connection.privmsg(sender,"Mods online:")
+            mods = list(set(mods)-set(ignorelist))
+            for mod in mods:
+                self.connection.privmsg(sender,mod)
+        else:
+            self.connection.privmsg(sender,"No mods online :(")
+
+    def trainers_Msg(self,sender):
+        """
+        !trainers - private messages sender a list of online trainers
         """
-        print "got welcomed"
-        #self.connection.join("#aeolus")
+        query = QtSql.QSqlQuery(self.db)
+        query.prepare("select l.login from login as l inner join avatars as a on l.id = a.idUser where a.idAvatar = 62")
+        query.exec_()
+        trainerList = [] 
+        while(query.next()):
+            trainerList.append(query.value(0))
+        trainerList = map(strings.IRCFoldedCase,sorted(trainerList))
+        userList = sorted(self.channels.items()[0][1].users())
+        trainerList = list(set(userList) & set(trainerList))
+        if trainerList:
+            self.connection.privmsg(sender, "Trainers online:")
+            for trainer in trainerList:
+                self.connection.privmsg(sender, trainer)
+        else:
+            self.connection.privmsg(sender, "No trainers online :(")
+
+    def help_Msg(self,sender):
+        """!help - Displays a list of fafbot commands"""
+        #I did tests with Vee, my way is 2 seconds, your way is 4 seconds
```

Comment:
@Cheyans This file ought to contain Python, not some chat between you and someone else. Also note how one can not deduce who and what the "I", the "your" and the mentioned approaches are.